### ISSUE TITLE
Data Clone Error on message including Notifications objects

### ISSUE THREAD

**Issue by dhowe** [dhowe/AdNauseam#1163]
```
[firefox/index.js][debug] Firefox stdout: Extension error: DataCloneError: The object could not be cloned. undefined 700
[[Exception stack
CallbackWrapper.prototype.proxy@moz-extension://2e2b68e8-e058-c84e-bbaf-484fa29d89d2/js/vapi-background.js:700:13
@moz-extension://2e2b68e8-e058-c84e-bbaf-484fa29d89d2/js/adn/core.js:2040:7
vAPI.messaging.onPortMessage@moz-extension://2e2b68e8-e058-c84e-bbaf-484fa29d89d2/js/vapi-background.js:807:17

```

I believe this is the reason for all the settings and menu issues in WebExt and is caused by passing notifications back and forth with functions as properties, which is disallowed according to the [structured clone algorithm](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm)

**Comment by dhowe** [comment]
I believe this is the reason for all the settings and menu issues and is caused by passing notifications back and forth with functions as properties, which is disallowed according to the [structured clone algorithm](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm)

**Comment by dhowe** [comment]
as a temporary hack I've added a function to simply strip the function from the notification...

**Comment by dhowe** [comment]
fixed with 4316fe0ed
